Reader Comments: Follow conversation via notifications

dvdchr opened the pull request:
Refs: `pctCYC-9x-p2`
Depends on #17369

This change enables the "Follow Conversation via Notifications" feature flag for the public. Additionally, this PR also addresses https://github.com/wordpress-mobile/WordPress-iOS/pull/17339#discussion_r732240314 regarding localization strings placements.

For now, the strings are encapsulated in an enum in Swift, and exposed to the Obj-C side. Since the feature flag is not yet removed, it only returns the titles for enabling/disabling notifications. Once the feature flag is removed, I'll add the titles for following/unfollowing conversations.

## To test

- Go to Reader, and select any unsubscribed post.
- Verify that the "Follow" button should be visible on the navigation bar.
- Tap the Follow button.
- Verify that the snackbar with the "Enable" action button is shown.
- Tap on the "Enable" button.
- Verify that a feedback notice with an "Undo" action button is shown.
- Tap on the "Undo" action button.
- Verify that a feedback notice is shown.
- Verify that the Follow button has turned into a bell icon button in the navigation bar.
- Go to Reader, and select any subscribed post.
- Tap on the bell icon button.
- Verify that the bottom sheet is shown.
- Tap on the switch.
- Dismiss the bottom sheet.
- Verify that a feedback notification is shown, and correctly represents the desired state.
- Tap the bell icon button again.
- Tap on the Unfollow Conversation button.
- Verify that a feedback notification is shown.
- Verify that the bell icon button has now turned back into "Follow".

## Related PRs:
- #17369
- #17340 
- #17339 
- #17333
- #17322
- https://github.com/wordpress-mobile/WordPressKit-iOS/pull/452

## Regression Notes
1. Potential unintended areas of impact
When updating from the previous version, the "follow conversation" state may be incorrectly represented (e.g. off when it's supposed to be on). Otherwise, the feature itself is pretty isolated so it should not impact other features.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the feature.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests for the service changes related to the follow conversation feature.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

dvdchr committed:
Refactor strings related to post subscription notifications

dvdchr committed:
Enable follow conversation flag

dvdchr committed:
Update release notes

ScoutHarris commented on RELEASE-NOTES.txt:
```diff
@@ -1,6 +1,6 @@
 18.6
 -----
-
+* [*] Comments: Users can now follow conversation via notifications, in addition to emails. [#17363]
```
Personally, I think this is worthy of at least two stars to be sure it gets in the release notes. My 2 cents. 😄

dvdchr replied:
Done in https://github.com/wordpress-mobile/WordPress-iOS/pull/17363/commits/c1bbaf642c57fb53d2de86f372e776e439c37a11. 🙂

ScoutHarris reviewed (approved):
> Additionally, this PR also addresses #17339 (comment) regarding localization strings placements.

OMG it's so much better! 😄 Thank you!

And all the tests checked out.

:shipit:

ScoutHarris commented:
Hey @dvdchr . I see a `reader_toggle_follow_conversation` track when following a conversation is toggled, but not when `Enable` or `Undo` is selected on the snackbar, or in-app notifications is toggled via the sheet. Maybe we should add some tracking before release?

dvdchr committed:
Merge branch 'feature/follow-conversation-notification-tracks' into feature/enable-follow-conversation

dvdchr committed:
Update release notes

dvdchr commented:
> I see a `reader_toggle_follow_conversation` track when following a conversation is toggled, but not when `Enable` or `Undo` is selected on the snackbar, or in-app notifications is toggled via the sheet. Maybe we should add some tracking before release?

Hey @ScoutHarris! Good catch 👍 . I've added tracks through a different PR: #17369, and changed this PR's base branch to point to that branch.

dvdchr committed:
Merge remote-tracking branch 'origin/develop' into feature/enable-follow-conversation
